app/controllers/returns/ReturnsCheckYourAnswersController.scala

```diff
@@ -77,4 +68,42 @@ class ReturnsCheckYourAnswersController @Inject()(
         }
     }
 
+  private def calCalculationAndCreditApi(
+                                          request: DataRequest[_]
+                                        )
+                                        (implicit hc: HeaderCarrier): Future[(Calculations, Credits)] = {
+    val fCalculations = returnsConnector.getCalculationReturns(request.pptReference)
+    val fCredits = getCredits(request)
+
+    for {
+      calculations <- fCalculations
+      credits <- fCredits
+    } yield (calculations, credits) match {
+      case (Right(calculations), Right(credits)) => (calculations, credits)
+      case _ => throw new RuntimeException("Error: There was a problem retrieving return calculation or the credits balance")
+    }
+  }
+  private def displayPage(request: DataRequest[_], obligation: TaxReturnObligation)
+                         (implicit messages: Messages, hc: HeaderCarrier): Future[Result] = {
+
+    calCalculationAndCreditApi(request).map {
+      case (calculations, credits) =>
+        val returnViewModel = TaxReturnViewModel(request, obligation, calculations)
+
+        Ok(view(returnViewModel, credits)(request, messages))
+    }
+  }
+
+  private def getCredits(request: DataRequest[_])
+                        (implicit hc: HeaderCarrier): Future[Either[ServiceError, Credits]] = {
+    if(request.userAnswers.getOrFail(WhatDoYouWantToDoPage)) {
+        creditsCalculatorConnector.get(request.pptReference).map {
+          case Right(creditBalance) => Right(CreditsClaimedDetails(request.userAnswers, creditBalance = creditBalance))
```

Comment:
Does this need to wrap up with user answers? The calling method already has useranswers as it provided the request